invalid address csr updates (#456)
riscv/riscv-cheri: src/riscv-integration.adoc

{cheri_base_ext_name} when accessing CSRs, branching and jumping, and
accessing memory.

===== Updating CSRs

The procedure below must be used when update a CSR with a capability A, and the CSR cannot
hold all invalid addresses, under the following circumstances:

* executing instructions, such as <<CSRRW>>
* hardware updates to CSRs such as storing the <<pcc>> (which becomes capability A) into
 <<mepcc>>/<<sepcc>> etc. when taking an exception.

The procedure is:

. If A's address is invalid, A does not have infinite bounds (see
xref:section_cap_encoding[xrefstyle=short]), and A's address has been modified due

> to illegal address conversion, then A's tag is set to 0.

Reviewer: ```suggestion
to invalid address conversion, then A's tag is set to 0.
```
"Invalid address" is the term used throughout here, rather than "illegal address".

Author: yeah - and that's what's in the text - so I'm a bit confused - maybe that was the previous commit

{cheri_base_ext_name} when accessing CSRs, branching and jumping, and
accessing memory.

===== Updating CSRs

> The procedure below must be used when update a CSR with a capability A, and the CSR cannot
> hold all invalid addresses, under the following circumstances:

Reviewer 2: ```suggestion
Some capability-holding CSRs are unable to hold arbitrary addresses.
For a given such CSR, addresses that cannot be stored are termed "invalid".
Some situations may appear to require that the CSR come to hold a capability with an invalid address:
```

* executing instructions, such as <<CSRRW>>
* hardware updates to CSRs such as storing the <<pcc>> (which becomes capability A) into
 <<mepcc>>/<<sepcc>> etc. when taking an exception.

> The procedure is:
> 
> . If A's address is invalid, A does not have infinite bounds (see

Reviewer 2: ```suggestion
In order to satisfy the definitions of such CSRs and preserve capability system invariants, the following procedure must be used as part of write-back to the CSR:

. If A's address is invalid and A does not have infinite bounds (see
```
